src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -331,14 +334,22 @@ public function denormalize($data, string $type, string $format = null, array $c
                 }
             }
 
-            $value = $this->validateAndDenormalize($resolvedClass, $attribute, $value, $format, $context);
             try {
-                $this->setAttributeValue($object, $attribute, $value, $format, $context);
-            } catch (InvalidArgumentException $e) {
-                throw new NotNormalizableValueException(sprintf('Failed to denormalize attribute "%s" value for class "%s": '.$e->getMessage(), $attribute, $type), $e->getCode(), $e);
+                $denormalizedValue = $this->validateAndDenormalize($resolvedClass, $attribute, $value, $format, $context);
+                try {
+                    $this->setAttributeValue($object, $attribute, $denormalizedValue, $format, $context);
+                } catch (InvalidArgumentException $e) {
+                    throw new NotNormalizableValueException(sprintf('Failed to denormalize attribute "%s" value for class "%s": '.$e->getMessage(), $attribute, $type), $e->getCode(), $e);
+                }
```

Comment:
I don't know the internals well here. Do we need BOTH the `$this->validateAndDenormalize()` and `$this->setAttributeValue()` to be inside the try-catch for `InvariantViolationException`?

Reply:
Actually if we can denormalize the value (which means it is valid) but can't write it, I don't think this is an invariant violation. I moved that parted out of the new try-catch.